### ISSUE TITLE
"Go to Definition" and "Find All References" don't work for constructors

### ISSUE THREAD

**Issue by m4tiz5zktonj** [markw65/prettier-extension-monkeyc#22]
Navigation to definition for `new MyObject()` opens class itself and not the `initialize` method in it. Similarly, there are no usages for `initialize` method are found despite having object instance creation expressions.

**Comment by markw65** [comment]
Fixed by fd45eb01ff0dcf4b214b3f49d2ce1cf47a64acc1, together with some changes to `@markw65/monkeyc-optimizer`

**Comment by markw65** [comment]
While I was working on the last release, I found an issue which probably explains what's going on here. In most circumstances, when trying to determine what x.B refers to, when x is known to be an object of class A, it might refer to any B defined in any sub-class of A, in addition to B's found in A or super-classes of A.

But A.B will only ever find a B defined in A, or a super-class of A.

Essentially I need to differentiate between x is an instance of A (so it might be a sub-class) and x is definitely an A and not a sub-class. I started to fix that, but decided it was too big a change, as I wanted to get the release out, and didn't actually have any examples of it causing problems.

I'll take another look...
